parsigs/parse_sig_api.py

```diff
@@ -87,26 +88,39 @@ def _parse_sig(sig: str, model: Language):
     return _create_structured_sigs(model_output)
 
 
-def _pre_process(sig):
-    sig = sig.lower().replace('twice', '2 times').replace("once", '1 time').replace("nightly", "every night")
+def _autocorrect(sig):
+    spell = SpellChecker()
+    spell.word_frequency.load_text_file(os.path.join('data', 'drug_names.txt'))
+    sig = sig.lower().strip()
+    corrected_words = []
+    for word in sig.split():
+        if not re.match(r"^[a-zA-Z]+$",word) or spell.known([word]):
+            corrected_words.append(word)
+        else:
+            corrected_word = spell.correction(word)
+            corrected_words.append(corrected_word)
+    sig = ' '.join(corrected_words)
+    return sig
 
-    sig = _add_space_around_parentheses(sig)
 
+def _pre_process(sig):
+    sig = _autocorrect(sig)
+    replacements = {
+            '@': 'a', 
```

Comment:
@itay-goldraich you have not returned this to previous logic, please do so we can merge